apm-agent-core/src/main/java/co/elastic/apm/agent/bci/IndyBootstrap.java

```diff
@@ -430,7 +431,12 @@ private static ConstantCallSite internalBootstrap(MethodHandles.Lookup lookup, S
                 // avoid re-entrancy and stack overflow errors
                 // may happen when bootstrapping an instrumentation that also gets triggered during the bootstrap
                 // for example, adding correlation ids to the thread context when executing logger.debug.
-                logger().warn("Nested instrumented invokedynamic instruction linkage detected", new Throwable());
+                if (!loggingForNestedCall.isNestedCallAndIncrement()) {
```

Comment:
This caused the tests to timeout due to not making progress. This only happens due to our test instrumentation where instrument `java.lang.System` to return custom environment variable values.